### PR TITLE
optimize AVX backend's accumulate_dw function

### PR DESCRIPTION
**Pull request by beru**
Hi, this PR optimizes AVX backend.

### CODE REVIEW

File: tiny_dnn/core/kernels/avx_kernel_common.h [hunk 1]
```diff
@@ -36,22 +36,46 @@
 #endif
 
 inline __m256 madd256_ps(__m256 a, __m256 b, __m256 c) {
+#ifdef CNN_USE_AVX2
```
Comment (Randl) [inline tiny_dnn/core/kernels/avx_kernel_common.h]: Putting single `#ifdef CNN_USE_AVX2` around all the code should be more readable

**Comment by Randl** [comment]
I'm sure you are aware of what you're doing, but what is speedup of this?


@edgarriba @nyanp We should start benchmarking both the master branch and PRs.

File: tiny_dnn/core/kernels/avx_kernel_common.h (hunk 1 above)
Comment (beru) [inline tiny_dnn/core/kernels/avx_kernel_common.h]: Thanks for the advice!

**Comment by beru** [comment]
> I'm sure you are aware of what you're doing, but what is speedup of this?

About 13% speed increase.

Tested with `example_cifar_train` built with Visual Studio 2015 on Windows10.
CPU : Intel Core i5 4670
Memory : DDR3 PC3-12800 Dual Channel

Current master branch (7fe3c7f8b21f2250e153a4d21c55eecddaa5dd30)
```
learning rate:0.01
load models...
start learning

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
84.3245s elapsed.
4793/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
83.9455s elapsed.
5166/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
83.9138s elapsed.
5464/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
83.81s elapsed.
5737/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
83.8489s elapsed.
5942/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
*****
```

PR branch (7b43aec3923584a0ea50f9772233e531b9c14688)
```
learning rate:0.01
load models...
start learning

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
73.5617s elapsed.
4790/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
73.1646s elapsed.
5169/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
72.726s elapsed.
5469/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
73.2064s elapsed.
5725/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
73.0698s elapsed.
5922/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
****
```

PR branch (7b43aec3923584a0ea50f9772233e531b9c14688)
Modified `train.cpp`. `n_minibatch = 16`
```
learning rate:0.01
load models...
start learning

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
63.2047s elapsed.
4911/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
62.7725s elapsed.
5281/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
62.742s elapsed.
5600/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
62.7517s elapsed.
5825/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
***************************************************
62.4386s elapsed.
6016/10000

0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
*****
```

**Comment by edgarriba** [comment]
@Randl definitely we should

**Comment by edgarriba** [comment]
@beru thx for your contribution! :+1:

**Pull request merged**